src/Scripting/CSharpTest/ScriptTests.cs

```diff
@@ -179,6 +179,35 @@ public void Do()
         }
 
         [Fact]
+        public void TestRunEmbeddedStatementNotFollowedBySemicolon()
+        {
+            var exceptionThrown = false;
+
+            try
+            {
+                var state = CSharpScript.RunAsync(@"
+ if (true)
+ System.Console.WriteLine(true)", globals: new ScriptTests());
```

Comment:
>System.Console.WriteLine(true) [](start = 1, length = 30)

We should have a test for a similar scenario in a non-script context in a method body.